Too optimistic bound on ghc-events leads to build failure

After https://github.com/mpickering/hs-speedscope/commit/0eecaa3aa9f6ca26d50ee7e0ca57826dec4d36a9, the solver can find a plan to compile with GHC 9.10, but this is futile:

```
❯ cabal build
...
Downloading  ghc-events-0.20.0.0
...
Building library for hs-speedscope-0.2.1...
[1 of 2] Compiling Speedscope.Schema ( src/Speedscope/Schema.hs, dist-newstyle/build/x86_64-linux/ghc-9.10.1/hs-speedscope-0.2.1/build/Speedscope/Schema.o, dist-newstyle/build/x86_64-linux/ghc-9.10.1/hs-speedscope-0.2.1/build/Speedscope/Schema.dyn_o )
[2 of 2] Compiling HsSpeedscope     ( src/HsSpeedscope.hs, dist-newstyle/build/x86_64-linux/ghc-9.10.1/hs-speedscope-0.2.1/build/HsSpeedscope.o, dist-newstyle/build/x86_64-linux/ghc-9.10.1/hs-speedscope-0.2.1/build/HsSpeedscope.dyn_o )
src/HsSpeedscope.hs:222:36: error: [GHC-83865]
    • Couldn't match type ‘Word16’ with ‘Word32’
      Expected: Capset
        Actual: ghc-events-0.20.0.0:GHC.RTS.EventTypes.CapNo
    • In the first argument of ‘Sample’, namely ‘t’
      In the first argument of ‘(:)’, namely ‘Sample t (V.toList st)’
      In the expression: Sample t (V.toList st) : el_samples elProf
    |
222 |       elProf { el_samples = Sample t (V.toList st) : el_samples elProf }
    |                                    ^

Error: [Cabal-7125]
Failed to build hs-speedscope-0.2.1 (which is required by exe:hs-speedscope from hs-speedscope-0.2.1).
```

cc @TeofilC

--- a/src/HsSpeedscope.hs
+++ b/src/HsSpeedscope.hs
@@ -219,7 +219,7 @@ processEventsDefault elProf (Event _t ei _c) =
     HeapProfCostCentre n l m s _ ->
       elProf { cost_centres = CostCentre n l m s : cost_centres elProf }
     ProfSampleCostCentre t _ _ st ->
-      elProf { el_samples = Sample t (V.toList st) : el_samples elProf }
+      elProf { el_samples = Sample (fromIntegral t) (V.toList st) : el_samples elProf }
     _ ->
       elProf
 

